tests/test_db_manager.py

```diff
@@ -0,0 +1,165 @@
+                    color_name VARCHAR NOT NULL
+                );"""
+
+        db.execute_simple(sql)
+        sql = f"""
+        insert into public.{test_table_name}(color_name)
+        values ('red'),('green'),('pink'),('purple'),('blue'),('cyan')
+        """
+        db.execute_simple(sql)
+
+        return test_table_name
+
+    # def test__get_connection(self):
+    #     db = self._get_db_inst()
+    #     conn = db._get_connection()
+    #     self.assertEqual(0, conn.closed)
+    #     # db._safe_tunnel_close(conn)
+
+    def test__get_sql_dataframe(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+
+        res = db.get_sql_dataframe(f'select * from {table_name}')
+        golden = 'red'
+        test = res.iloc[0]['color_name']
+        self.assertEqual(golden, test)
+
+        # @todo test with params.
+
+    def test__get_sql_list_dicts(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+
+        res = db.get_sql_list_dicts(f'select * from {table_name}')
+        golden = 'red'
+        test = res[0]['color_name']
+        self.assertEqual(golden, test)
+
+    def test__get_sql_single_item_list(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+        res = db.get_sql_single_item_list(f'select color_name from {table_name}')
+        golden = 'red'
+        test = res[0]
+        self.assertEqual(golden, test)
+
+    def test__execute_simple(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+        sql = f"""
+        insert into public.{table_name}(color_name)
+        values ('abc')
+        """
+        db.execute_simple(sql)
+        res = db.get_sql_single_item_list(f'select color_name from {table_name}')
+        golden = 'abc'
+        test = res[len(res)-1]
+        self.assertEqual(golden, test)
+
+    def test__get_single_result(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+        res = db.get_single_result(f'select color_name from {table_name}')
+        golden = 'red'
+        test = res[len(res)-1]
+        self.assertEqual(golden, test)
+
+    def test__execute_many(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+
+        # Test normal function
+        sql = 'insert into public.color (color_name) values %s;'
+        params = [['ivory'], ['lemon'], ['copper'], ['salmon'], ['rust'], ['amber'], ['cream'], ['tan'], ['bronze'], ['blue'], ['silver'], ['grey']]
+        db.execute_many(sql, params)
+
+        res = db.get_sql_single_item_list(f'select color_name from {table_name}')
+
+        golden = 'grey'
+        test = res[len(res)-1]
+        self.assertEqual(golden, test)
+
+        # Test errors - Bad column name
+        sql = 'insert into public.color (column_that_doesnt_exist) values %s;'
+        failed = False
+        try:
+            db.execute_many(sql, params)
```

Comment:
You can use `assertRaises` to check that the expected exception is raised, rather than setting a flag in a try/except

Reply:
I was having some real issues with assertRaises. I will give it another shot.

Reply:
Ok, if you see issues again can you post the error messages here?